hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -371,6 +378,35 @@ private Set<String> getSSTFileListForSnapshot(OmSnapshot snapshot,
         .getPath(), tablesToLookUp);
   }
 
+  public List<SnapshotDiffJob> getSnapshotDiffJobList(
+      String volumeName, String bucketName,
+      String jobStatus, boolean listAll) {
+    List<SnapshotDiffJob> jobList = new ArrayList<>();
+
+    try (ClosableIterator<Map.Entry<String, SnapshotDiffJob>> iterator =
+             snapDiffJobTable.iterator()) {
+      while (iterator.hasNext()) {
+        SnapshotDiffJob snapshotDiffJob = iterator.next().getValue();
+        if (snapshotDiffJob.getVolume().equals(volumeName) &&
+            snapshotDiffJob.getBucket().equals(bucketName)) {
+          if (listAll) {
+            jobList.add(snapshotDiffJob);
+            continue;
+          }
+
+          // If provided job status is invalid,
+          // then all jobs on the table will be ignored.
+          // No need to check if getJobStatusFromString doesn't return null.
+          if (snapshotDiffJob.getStatus().equals(
```

Comment:
It is better to use `==` for enum to be null safe.

https://stackoverflow.com/a/1750453

Reply:
> Alternative, could use [Objects.equals()](https://github.com/openjdk/jdk/blob/jdk8-b120/jdk/src/share/classes/java/util/Objects.java#L43-L60)?

That works as well.